modules/videoio/src/cap_ffmpeg_hw.hpp

```diff
@@ -80,7 +113,7 @@ std::string getDecoderConfiguration(VideoAccelerationType va_type, AVDictionary
     case VIDEO_ACCELERATION_ANY: return "vaapi.iHD";
     case VIDEO_ACCELERATION_D3D11: return "";
     case VIDEO_ACCELERATION_VAAPI: return "vaapi.iHD";
-    case VIDEO_ACCELERATION_MFX: return "";
+    case VIDEO_ACCELERATION_MFX: return "qsv.iHD";
```

Comment:
No easy workaround. ffmpeg-qsv decoder designed with delayed initialization, the `MFXVideoDECODE_Init` called on decoding first frame, it's too late for fallback to SW decoder in current implementation. No such problem in VAAPI and D3D11 decoders as they report all errors on opening stage (av_codec_open2 call).
Disable MFX decode (revert "qsv.iHD" back to empty string "") or keep as known issue in case MFX acceleration explicitly requested?

Reply:
@mikhail-nikolskiy Thank you for the information!

Looks like we need to add the first frame grabbing as part of `.open()` implementation to handle/workaround this case (I believe we can implement this later).